component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -821,6 +821,26 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             }
             context.setProperty(SMSOTPConstants.CODE_MISMATCH, true);
         }
+
+        if (succeededAttempt && isLocalUser) {
+            String username = String.valueOf(context.getProperty(SMSOTPConstants.USER_NAME));
+            try {
+                String mobileNumber = SMSOTPUtils.getMobileNumberForUsername(username);
+                if (StringUtils.isEmpty(mobileNumber)) {
+                    String tenantDomain = context.getTenantDomain();
+                    if (!tenantDomain.equals(SMSOTPConstants.SUPER_TENANT)) {
+                        IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, getName(), tenantDomain);
```

Comment:
Is this part required here?

Reply:
Tested without that code segment and it is working fine. Updated the pull request with latest code